src/transaction-validator.ts

```diff
@@ -2,50 +2,111 @@ import { Transaction, TransactionInput } from './types';
 import { UTXOPoolManager } from './utxo-pool';
 import { verify } from './utils/crypto';
 import {
-  ValidationResult,
-  ValidationError,
-  VALIDATION_ERRORS,
-  createValidationError
+	ValidationResult,
+	ValidationError,
+	VALIDATION_ERRORS,
+	createValidationError
 } from './errors';
 
 export class TransactionValidator {
-  constructor(private utxoPool: UTXOPoolManager) {}
+	constructor(private utxoPool: UTXOPoolManager) {}
 
-  /**
+	/**
    * Validate a transaction
    * @param {Transaction} transaction - The transaction to validate
    * @returns {ValidationResult} The validation result
    */
-  validateTransaction(transaction: Transaction): ValidationResult {
-    const errors: ValidationError[] = [];
+	validateTransaction(transaction: Transaction): ValidationResult {
+		const errors: ValidationError[] = [];
+		let inputAmount = 0;
+		let outputAmount = 0;
 
-    // STUDENT ASSIGNMENT: Implement the validation logic above
-    // Remove this line and implement the actual validation
-    throw new Error('Transaction validation not implemented - this is your assignment!');
+		if(transaction.inputs.length === 0){
+			errors.push(createValidationError(
+				VALIDATION_ERRORS.EMPTY_INPUTS,
+				'Empty amount of inputs' )
+			);
+		}
+		if(transaction.outputs.length === 0){
+			errors.push(createValidationError(
+				VALIDATION_ERRORS.EMPTY_OUTPUTS,
+				'Empty amount of outputs' )
+			);
+		}
 
-    return {
-      valid: errors.length === 0,
-      errors
-    };
-  }
+		const usedUtxos = new UTXOPoolManager();
 
-  /**
+		for(const input of transaction.inputs){
+			const anUTXO = this.utxoPool.getUTXO(input.utxoId.txId, input.utxoId.outputIndex);
+			if(!anUTXO){
+				errors.push(createValidationError(
+					VALIDATION_ERRORS.UTXO_NOT_FOUND,
+					'UTXO not found: ${input.utxoId.txId}:${input.utxoId.outputIndex}')
+				);
+			}
+			if(anUTXO){
```

Comment:
No es necesario colocar este `if`. Si tu intención era que el código en este statement no se ejecutara, se puede usar `continue` en el statement anterior, para saltar a la siguiente iteración del `for`. Pero más allá de eso... En todo caso podés usar un `else`.